pennylane/devices/qubit/preprocess.py

```diff
@@ -95,34 +100,18 @@ def validate_and_expand_adjoint(
         Union[.QuantumTape, .DeviceError]: The expanded tape, such that it is supported by adjoint differentiation.
         If the circuit is invalid for adjoint differentiation, a DeviceError with an explanation is returned instead.
     """
```

Comment:
we should add "Raises: DeviceError" to the docstring. Also, it's starting to get confusing now that it both raises and returns DeviceErrors. Should we always just return it?

Reply:
Given that the recursion error is enough an extreme edge case, I'm not too concerned.